assembl/auth/mediactive.py

```diff
@@ -0,0 +1,43 @@
+from urllib import urlencode
+from HTMLParser import HTMLParser
+
+from simplejson import loads
+from social.backends.base import BaseAuth
+
+from .encryption import MediactiveAESCryptor
+
+
+class Mediactive(BaseAuth):
+    """A simple SocialAuth backend agreed to with Mediactive.
+    Security is based on a shared secret key."""
+    name = 'mediactive'
+    USERNAME_KEY = 'username'
+    html_parser = HTMLParser()
+
+    def auth_url(self):
+        """Must return redirect URL to auth provider"""
+        next_url = self.data.get('next', None)
+        base = self.setting('LOGIN_URL')
+        if next_url:
+            base += '?' + urlencode({'next': next_url})
```

Comment:
Use `urljoin` and `urlencode` to ensure flexible usage of the config file under the `LOGIN_URL` key.

Reply:
urlencode I'm using... not sure what urljoin is adding anything to this. 
Unless you mean urlparse/urldefrag?

Reply:
I wrote it for the heck of it, but I think it's overkill. Tell me if you want it.
```python
        if next_url:
            pr = urlparse(base)
            qs = parse_qs(pr.query)
            qs['next'] = [next_url]
            base = urlunparse(pr._replace(query=urlencode(qs, True)))
```